src/components/application_manager/src/application_manager_impl.cc

```diff
@@ -2444,6 +2453,26 @@ void ApplicationManagerImpl::HeadUnitReset(
   }
 }
 
+void ApplicationManagerImpl::ClearAppsPersistentData() {
+  const std::string apps_info_storage_file = get_settings().app_info_storage();
+  file_system::DeleteFile(apps_info_storage_file);
+
+  const ApplicationSet& accessor = applications().GetData();
+
+  ApplicationSetConstIt it_app_list = accessor.begin();
```

Comment:
@AByzhynar 
Is this logic applicable only for registered applications?
If not (i.e. for all even non-registered apps) than same logic is in MASTER_RESET and you can use one method for that.